src/game/common/audio/audioeventrts.cpp

```diff
@@ -77,7 +75,15 @@ AudioEventRTS::AudioEventRTS(const AudioEventRTS &that) :
     m_playerIndex(that.m_playerIndex),
     m_nextPlayPortion(that.m_nextPlayPortion)
 {
-    m_positionOfAudio.Set(&that.m_positionOfAudio);
+    if (m_eventType != EVENT_MUSIC && m_eventType != EVENT_UNKVAL3) {
+        if (m_eventType == EVENT_SPEECH) {
+            m_drawableID = that.m_drawableID;
```

Comment:
This does not look proper, even if original is like so.

Essentially this class instance will keep old ID's after copy, if code path enables it so.

Meaning if `m_eventType` is not `EVENT_SPEECH`, then `m_drawableID` is not reset to invalid drawable id. Maybe this could lead to bugs? Same for `m_objectID` and `m_positionOfAudio`.

Reply:
This is 100% correct, m_objectID and m_drawableID are a union on top of each other so the way this works is how its supposed to work.